scripts/upgrade-droplets.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+cat <<-HELPTEXT
+
+ Upgrade/install the latest do-agent on all of the accounts droplets.
+ You must provide a DO API token as AUTH_TOKEN environment variable.
+
+ Example: AUTH_TOKEN=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX ./upgrade-droplets.sh install
+
+HELPTEXT
+
+set -ue
+
+function main() {
+	cmd=${1:-}
+	[ -z "$cmd" ] && usage && exit 0
+	shift
+	fn=command_$cmd
+	# disable requirement to quote 'fn' which would break this code
+	# shellcheck disable=SC2086
+	if [ "$(type -t ${fn})" = function ]; then
+		${fn} "$@"
+	else
+		usage
+		exit 1
+	fi
+}
+
+# show the help text
+function command_help() {
+	usage
+}
+
+# show script usage. This parses every function named command_ from this script
+# and displays it as a possible command
+function usage() {
+	echo
+	echo "Usage: $0 [command]"
+	echo
+	echo "Possible commands: "
+	grep -P '^function command_' "$0" \
+		| sed 's,function command_,,g' \
+		| sed 's,() {,,g' \
+		| sort \
+		| xargs -n1 -I{} echo "  {}"
+	echo
+}
+
+# execute a command against a list of IP addresses
+# Usage:   exec_ips <ips> <command>
+# Example: exec_ips "$(list_ips_rpm)" "yum update DO agent"
+function exec_ips() {
+	{ [ -z "${1:-}" ] || [ -z "${2:-}" ]; } \
+		&& abort "Usage: ${FUNCNAME[0]} <ips> <command>"
+
+	ips=$1
+	shift
+	script="hostname -s; { $*; }"
+	echo "================================================"
+	echo "> $*"
+	echo "================================================"
+	for ip in $ips; do
+		# shellcheck disable=SC2029
+		echo "$(echo
+			echo -n "$(tput setaf 2)>>>> $ip: $(tput sgr 0)"
+			ssh -o "StrictHostKeyChecking no" \
+				-o "UserKnownHostsFile=/dev/null" \
+				-o "LogLevel=ERROR" \
+				"root@${ip}" "${script}" 2>/dev/stdout || true
```

Comment:
Lets create another global variable called `SSH_KEY=${SSH_KEY:-~/.ssh/id_rsa}` and add a flag here `-i "${SSH_KEY}` so the user can specify their key to use if they want

Reply:
done